tests/test_seasonal_date_factor.R

```diff
@@ -0,0 +1,120 @@
+    checkTrue((as.integer(format(ci.csv@dates[length(ci.csv@dates)], "%m")) + 1 %% 3 != 0) && is.na(ci.csv@namasks$seasonal[[var]][length(ci.csv@namasks$season[[var]])]))
+    # Test 1.3: Check for NA seasons where conditions in 1.1 and 1.2 do not apply
+    # A season is marked as NA if it contains at least one month with missing data or if the sum of missing days within the season is greater than or equal to max.missing.days$season.
+    unique_seasons <- unique(ci.csv@date.factors$seasonal)
+    unique_months <- unique(ci.csv@date.factors$monthly)
+    na_seasons <- unique_seasons[is.na(ci.csv@namasks$seasonal[[var]])]
+    na_months <- unique_months[is.na(ci.csv@namasks$monthly[[var]])]
+    results <- logical(length(na_seasons))
+    
+    for (i in seq_along(na_seasons)) {
+      season <- na_seasons[i]
+      months_of_season <- unique(ci.csv@date.factors$monthly[ci.csv@date.factors$season == season])
+      any_month_na <- any(months_of_season %in% na_months)
+      results[i] <- any_month_na
+      na_days <- sum(is.na(ci.csv@data[[var]][ci.csv@date.factors$season == season]))
+      results[i] <- any_month_na || (na_days >= ci.csv@max.missing.days[["seasonal"]])
+      
+    }
+    checkTrue(all(results))
+  }
+  
+}
+
+
+climdex.pcic.test.seasonal.indices <- function() {
+  # Test 2:
+  # This section handles various scenarios based on the type of index being calculated:
+  # - For non-averaged stats (e.g., min, max), the seasonal value is the min/max of the constituent months.
+  # - For averaged stats, the seasonal value may differ from the average of the 3 constituent months due to varying month lengths, so it's compared against a calculated dtr (daily temperature range).
+  # - For percentile calculations, each season's percentile is computed against a weighted monthly percentile.
+
+  min_max <- list(
+    "tnn" = list(stat = "min", var = "tmin"), "tnx" = list(stat = "max", var = "tmin"),
+    "txn" = list(stat = "min", var = "tmax"), "txx" = list(stat = "max", var = "tmax"),
+    "rx1day" = list(stat = "max", var = "prec"), "rx5day" = list(stat = "max", var = "prec")
+  )
+  mean_indices <- list("dtr" = list(var = c("tmin", "tmax")))
+  bootstrap_indices <- list("tn10p" = list(var = "tmin"), "tn90p" = list(var = "tmin"), "tx10p" = list(var = "tmax"), "tx90p" = list(var = "tmax"))
+  seasonal_indices <- c(names(min_max), names(mean_indices), names(bootstrap_indices))
+  ci_subset <- function(ci.csv, dates, season) {
+    return(climdexInput.raw(
+      tmax = ci.csv@data$tmax[ci.csv@date.factors$seasonal == season],
+      tmin = ci.csv@data$tmin[ci.csv@date.factors$seasonal == season],
+      prec = ci.csv@data$prec[ci.csv@date.factors$seasonal == season],
+      tavg = ci.csv@data$tmin[ci.csv@date.factors$seasonal == season],
+      tmax.dates = dates, tmin.dates = dates, prec.dates = dates, tavg.dates = dates, base.range = c(1981, 1990)
+    ))
+  }
+  for (index in seasonal_indices) {
```

Comment:
I'm not keen on the structure of this loop. While it covers a lot of different cases, there is too much nested logic and switches to be able to easily follow the program flow. Can you break this up in a different way?

Reply:
Yes, I will break up the nested logic, and run separate tests for each type of index (min-max/mean/bootstrap).